packages/dev/deploy/deploy.ts

```diff
@@ -0,0 +1,178 @@
+// @ts-ignore
+import { ethers } from 'hardhat'
+
+const { AddressZero } = ethers.constants
+
+const deploymentFunc: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const deployTestPaymaster = true
+
+  // TODO: there should be type extensions to support these...
+  const { web3, deployments, getChainId } = hre as any
+  const accounts = await ethers.provider.listAccounts()
+  const deployer = accounts[0]
+
+  async function deploy (name: string, options: DeployOptions): Promise<DeployResult> {
+    console.log('Deploying: ', name)
+    const res = await deployments.deploy(name, options)
+    console.log(name, res.address, res.newlyDeployed as boolean ? chalk.yellow('newlyDeployed') : chalk.gray('existing'))
+    return res
+  }
+
+  const setField = async function (contract: string, getFunc: string, setFunc: string, val: any, options = {
+    from: deployer,
+    log: true
+  }): Promise<void> {
+    const currentVal = await deployments.read(contract, options, getFunc)
+    if (currentVal !== val) {
+      await deployments.execute(contract, options, setFunc, val)
+    }
+  }
+
+  const balance = await ethers.provider.getBalance(deployer)
+  console.log('deployer=', deployer, 'balance=', formatEther(balance.toString()))
+  // if (balance.isZero()) {
+  //   console.error(chalk.red(`deployer account ${deployer as string} doesn't have eth balance ${balance.toString()}`))
+  //   process.exit(1)
+  // }
+
+  let isArbitrum = false
+  try {
+    const arbSysAddress = '0x' + '64'.padStart(40, '0')
+    const ArbSys = new ethers.Contract(arbSysAddress, ['function arbOSVersion() external pure returns (uint)'], ethers.provider)
+    const arbos = await ArbSys.arbOSVersion()
+
+    console.log('== Running on', chalk.yellowBright('Arbitrum'), 'arbOSVersion=', arbos)
+    isArbitrum = true
+  } catch (e) {
+  }
+
+  const chainId = parseInt(await getChainId())
+
+  const envname = Object.keys(environments).find(env => environments[env as EnvironmentsKeys].chainId === chainId)
+  console.log('loading env ( based on chainId', chainId, ')', envname ?? 'DefaultEnvironment')
+  const env = envname != null ? environments[envname as EnvironmentsKeys] : defaultEnvironment
+
+  let stakingTokenAddress = process.env.TOKEN
+  if (stakingTokenAddress == null) {
+    console.error('must specify TOKEN address (or "test" to deploy TestWeth')
+    process.exit(1)
+  }
+  let stakingTokenValue = process.env.TOKEN_STAKE
+
+  if (stakingTokenAddress === 'test') {
+    const TestWEth = await deploy('TestWEth', {
+      from: deployer
+    })
+    stakingTokenAddress = TestWEth.address
+    stakingTokenValue = stakingTokenValue ?? '0.1'
+  } else {
+    if (stakingTokenValue == null) {
+      console.error('must specify TOKEN_STAKE for staking token')
+      process.exit(1)
+    }
+    const token = new ethers.Contract(stakingTokenAddress,
+      ['function symbol() view returns (string)'], ethers.provider)
```

Comment:
I am not familiar with this syntax, what does this `symbol` mean here? You don't need the ABI to declare contract functions? This is nice.